decoder/vaapiDecoderJPEG.cpp

```diff
@@ -358,6 +360,39 @@ YamiStatus VaapiDecoderJPEG::decode(VideoDecodeBuffer* buffer)
     return m_impl->decode(buffer->data, buffer->size);
 }
 
+//get frame fourcc, return 0 for unsupport format
+static uint32_t getFourcc(const FrameHeader::Shared& frame)
+{
+    if (frame->components.size() != 3) {
+        ERROR("unsupported compoent size %d", (int)frame->components.size());
+        return 0;
+    }
+    int h1 = frame->components[0]->hSampleFactor;
+    int h2 = frame->components[1]->hSampleFactor;
+    int h3 = frame->components[2]->hSampleFactor;
+    int v1 = frame->components[0]->vSampleFactor;
+    int v2 = frame->components[1]->vSampleFactor;
+    int v3 = frame->components[2]->vSampleFactor;
+    if (h2 != h3 || v2 != v3) {
+        ERROR("unsupported format h2 = %d, h3 = %d, v2 = %d, v3 = %d", h2, h3, v2, v3);
+        return 0;
+    }
+    if (h1 == h2) {
+        if (v1 == v2)
+            return YAMI_FOURCC_444P;
+        if (v1 == 2 * v2)
+            return YAMI_FOURCC_422V;
+    }
+    else if (h1 == 2 * h2) {
+        if (v1 == v2)
+            return YAMI_FOURCC_422H;
+        if (v1 == 2 * v2)
+            return YAMI_FOURCC_IMC3;
```

Comment:
A unique `DEBUG` message before each return value to indicate which fourcc was detected might be nice here ;)